GoogleUtilities/Tests/Unit/Swizzler/GULAppDelegateSwizzlerTest.m

```diff
@@ -1087,7 +1087,135 @@ - (void)testApplicationDidReceiveRemoteNotificationWithCompletionIsInvokedOnInte
 
   XCTAssertEqual(testAppDelegate.application, application);
   XCTAssertEqual(testAppDelegate.remoteNotification, notification);
-  XCTAssertEqual(testAppDelegate.remoteNotificationCompletionHandler, completion);
+}
+
+- (void)testApplicationDidReceiveRemoteNotificationWithCompletionCompletionIsCalledOnce {
+  NSDictionary *notification = @{};
+  GULApplication *application = [GULApplication sharedApplication];
+
+  XCTestExpectation *completionExpectation =
+      [[XCTestExpectation alloc] initWithDescription:@"Completion called once"];
+  completionExpectation.assertForOverFulfill = YES;
+
+  void (^completion)(UIBackgroundFetchResult) = ^(UIBackgroundFetchResult result) {
+    [completionExpectation fulfill];
+  };
+
+  void (^onDidReceiveRemoteNotification)(NSInvocation *invocation) = ^(NSInvocation *invocation) {
+    void __unsafe_unretained (^localCompletionHandler)(UIBackgroundFetchResult) = nil;
+    [invocation getArgument:(void *)(&localCompletionHandler) atIndex:4];
+    XCTAssertNotNil(localCompletionHandler);
+    localCompletionHandler(UIBackgroundFetchResultNoData);
+  };
+
+  id interceptor = OCMProtocolMock(@protocol(GULApplicationDelegate));
+  OCMExpect([interceptor application:application
+                didReceiveRemoteNotification:notification
+                      fetchCompletionHandler:[OCMArg isNotNil]])
+      .andDo(onDidReceiveRemoteNotification);
+
+  id interceptor2 = OCMProtocolMock(@protocol(GULApplicationDelegate));
+  OCMExpect([interceptor2 application:application
+                didReceiveRemoteNotification:notification
+                      fetchCompletionHandler:[OCMArg isNotNil]])
+      .andDo(onDidReceiveRemoteNotification);
+
+  GULTestAppDelegate *testAppDelegate = [[GULTestAppDelegate alloc] init];
+  OCMStub([self.mockSharedApplication delegate]).andReturn(testAppDelegate);
+  [GULAppDelegateSwizzler proxyOriginalDelegateIncludingAPNSMethods];
+
+  [GULAppDelegateSwizzler registerAppDelegateInterceptor:interceptor];
+  [GULAppDelegateSwizzler registerAppDelegateInterceptor:interceptor2];
+
+  [testAppDelegate application:application
+      didReceiveRemoteNotification:notification
+            fetchCompletionHandler:completion];
+  testAppDelegate.remoteNotificationCompletionHandler(UIBackgroundFetchResultNoData);
+  OCMVerifyAll(interceptor);
+  OCMVerifyAll(interceptor2);
+  [self waitForExpectations:@[ completionExpectation ] timeout:0.1];
+}
+
+- (void)
```

Comment:
It would be nice to add tests for cases with different fetch results or at least add a `TODO` to add them.